WebHost.py

```diff
@@ -39,17 +42,54 @@ def get_app():
 
 def create_ordered_tutorials_file() -> typing.List[typing.Dict[str, typing.Any]]:
     import json
-
-    with open(Utils.local_path("WebHostLib", "static", "assets", "tutorial", "tutorials.json")) as source:
-        data = json.load(source)
-
-    data = sorted(data, key=lambda entry: entry["gameTitle"].lower())
-
-    folder = Utils.local_path("WebHostLib", "static", "generated")
-    os.makedirs(folder, exist_ok=True)
-    with open(os.path.join(folder, "tutorials.json"), "w") as target:
-        json.dump(data, target)
-    return data
+    import shutil
+    worlds = {}
+    data = []
+    for game, world in AutoWorldRegister.world_types.items():
+        if hasattr(world.web, 'tutorials'):
+            worlds[game] = world
+    for game, world in worlds.items():
+        # copy files from world's docs folder to the generated folder
+        source_path = Utils.local_path(os.path.dirname(sys.modules[world.__module__].__file__), 'docs')
+        target_path = Utils.local_path("WebHostLib", "static", "generated", "docs", game)
+        files = os.listdir(source_path)
+        for file in files:
+            os.makedirs(os.path.dirname(Utils.local_path(target_path, file)), exist_ok=True)
+            shutil.copyfile(Utils.local_path(source_path, file), Utils.local_path(target_path, file))
+        # build a json tutorial dict per game
+        game_data = {'gameTitle': game, 'tutorials': [{}]}
+        for tutorial in world.web.tutorials:
+            # build dict for the json file
+            current_tutorial = {
+                'name': tutorial.tutorial_name,
+                'description': tutorial.description,
+                'files': [{
+                    'language': tutorial.language,
+                    'filename': game + '/' + tutorial.file_name,
+                    'link': f'{game}/{tutorial.link}',
+                    'authors': tutorial.author
+                }]
+            }
+            # probably a cleaner way to do this but this adds tutorials covering the same topics of different languages together
+            added: bool = False
+            if 'name' in game_data['tutorials'][0]:
+                for guide in game_data['tutorials']:
+                    if tutorial.tutorial_name in guide['name']:
+                        guide['files'].append(current_tutorial['files'][0])
+                        added = True
+                if not added:
+                    game_data['tutorials'].append(current_tutorial)
+            else:
+                game_data['tutorials'][0] = current_tutorial
+        data.append(game_data)
+    with open(Utils.local_path("WebHostLib", "static", "generated", "tutorials.json"), 'w', encoding='utf-8-sig') as json_target:
+        generic_data = {}
+        for games in data:
```

Comment:
This for -> if confuses me quite a bit reading it. In which value/key are you looking for Archipelago? Why are you iterating over values?

Reply:
I'm iterating over the list of `gameTitles` for Archipelago so I can put it as the very first entry on the site. I was unsure if the generic world is going to *always* be called first so I did this to future proof it but if generic will always be first I can remove this loop and just pull it out then sort and reinsert.

Reply:
Then "gameTitles" should probably appear in the code somewhere, instead of iterating over .values(). Both for reabability and speed.